source/adiabatic_conditions/initial_profile.cc

```diff
@@ -99,21 +113,26 @@ namespace aspect
           // Handle the case that cp is zero (happens in simple Stokes test problems like sol_cx). By setting
           // 1/cp = 0.0 we will have a constant temperature profile with depth.
           const double one_over_cp = (out.specific_heat[0]>0.0) ? 1.0/out.specific_heat[0] : 0.0;
-          const double gravity = this->get_gravity_model().gravity_vector(representative_point).norm();
+          double gravity = backward_advection * this->get_gravity_model().gravity_vector(representative_point).norm();
 
           pressures[i] = pressures[i-1]
                          + density * gravity * delta_z;
           temperatures[i] = temperatures[i-1] * (1 +
                                                  alpha * gravity * delta_z * one_over_cp);
         }
 
-      Assert (*std::min_element (pressures.begin(), pressures.end()) >=
-              -std::numeric_limits<double>::epsilon() * pressures.size(),
-              ExcMessage("Adiabatic InitialProfile encountered a negative pressure of "
-                         + dealii::Utilities::to_string(*std::min_element (pressures.begin(), pressures.end()))));
-      Assert (*std::min_element (temperatures.begin(), temperatures.end()) >=
-              -std::numeric_limits<double>::epsilon() * temperatures.size(),
-              ExcMessage("Adiabatic InitialProfile encountered a negative temperature."));
+      // Only check for negative temperature and pressure if the gravity is pointing
+      // up or out.
+      if (backward_advection == 1)
```

Comment:
Are the values always negative when doing backward advection? If yes, could you Assert that?